core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -122,6 +132,86 @@ public int read(ByteBuffer buf, int off, int len) throws IOException {
     return totalBytesRead;
   }
 
+  private int bufferedRead(byte[] b, int off, int len, ReadType readType, long pos,
+                           Stopwatch stopwatch) throws IOException {
+    //hit or partially hit the in stream buffer
+    if (pos > mBufferStartOffset && pos < mBufferEndOffset) {
+      int lengthToReadFromBuffer = (int) Math.min(len,
+          mBufferEndOffset - pos);
+      System.arraycopy(mBuffer, (int) (pos - mBufferStartOffset),
+          b, off, lengthToReadFromBuffer);
+      return lengthToReadFromBuffer;
+    } else {
+      if (len >= mBufferSize) {
+        // Skip load to the in stream buffer if the data piece is larger than buffer size
+        return localCachedRead(b, off, len, readType, pos, stopwatch);
+      } else {
+        int bytesLoadToBuffer = (int) Math.min(mBufferSize, mStatus.getLength() - pos);
+        int bytesRead = localCachedRead(mBuffer, 0, bytesLoadToBuffer, readType, pos, stopwatch);
+        mBufferStartOffset = pos;
+        mBufferEndOffset = pos + bytesRead;
+        int dataReadFromBuffer = Math.min(bytesRead, len);
+        System.arraycopy(mBuffer, 0, b, off, dataReadFromBuffer);
+        MetricsSystem.meter(MetricKey.CLIENT_CACHE_BYTES_READ_IN_STREAM_BUFFER.getName())
```

Comment:
Make all the `MetricsSystem.xxx` a static member variable. Fetching the metric is actually a string construction / matching problem which is not exactly cheap.

Reply:
Good call,  but I think getName is ok, it's just a simple getter for the `mName`, the compiler will inline this call.   but the getMetricsName is using a regex matching, I will make all the getMetricsName call to static members.  what do you think?